### PR TITLE
Update wrapper and enable back all warnings as errors for Kotlin

### PR DESCRIPTION
**Pull request by asodja**
To dogfood https://github.com/gradle/gradle/pull/24626 and https://github.com/gradle/gradle/pull/24687.

Also remove rewrite for Kotlin assignment plugin, since 1.8.20 doesn't emit warning anymore.

Fixes https://github.com/gradle/gradle/issues/23650

### CODE REVIEW

**Comment by asodja** [comment]
It seems it won't work out of the box.

**Comment by asodja** [comment]
It seems it needs kotlin-dsl update... So it's not a late Friday job

**Comment by asodja** [comment]
@bot-gradle test this

**Comment by asodja** [comment]
@bot-gradle test this

File: build-logic-commons/gradle-plugin/src/main/kotlin/gradlebuild.build-logic.kotlin-dsl-gradle-plugin.gradle.kts [hunk 1]
```diff
@@ -34,8 +34,7 @@ dependencies {
 
 tasks.withType<KotlinCompile>().configureEach {
     compilerOptions {
-        // Can be set to true when assignment plugin becomes stable
-        allWarningsAsErrors = false
+        allWarningsAsErrors = true
```
Comment (eskatos) [inline build-logic-commons/gradle-plugin/src/main/kotlin/gradlebuild.build-logic.kotlin-dsl-gradle-plugin.gradle.kts:37]: :tada:

**Review by eskatos — APPROVED** [review]
It seems CI is choking on the unrelated wrapper validation issue fixed in
* https://github.com/gradle/gradle/pull/24745

**Comment by asodja** [comment]
@bot-gradle test this

**Comment by asodja** [comment]
@bot-gradle test this

**Comment by asodja** [comment]
@bot-gradle clean

**Comment by asodja** [comment]
@bot-gradle test and merge

**Pull request merged**